machine-learning/tutorials/SentimentAnalysis/Program.cs

```diff
@@ -3,102 +3,131 @@
+            // When you create a loader, you specify the schema by passing a class to the loader containing
+            // all the column names and their types. This is used to create the model, and train it. 
+            // Initialize our TextLoader
+            // <Snippet4>
+            _textLoader = mlContext.Data.TextReader(new TextLoader.Arguments()
+                                                {
+                                                    Separator = "tab",
+                                                    HasHeader = true,
+                                                    Column = new[]
+                                                                {
+                                                                  new TextLoader.Column("Label", DataKind.Bool, 0),
+                                                                  new TextLoader.Column("SentimentText", DataKind.Text, 1)
+                                                                }
+                                                }
+            );
+            // </Snippet4>
 
-        public static async Task<PredictionModel<SentimentData, SentimentPrediction>> Train()
-        {
-            // LearningPipeline allows you to add steps in order to keep everything together 
-            // during the learning process.  
             // <Snippet5>
-            var pipeline = new LearningPipeline();
+            var model = Train(mlContext, _trainDataPath);
             // </Snippet5>
 
-            // The TextLoader loads a dataset with comments and corresponding postive or negative sentiment. 
-            // When you create a loader, you specify the schema by passing a class to the loader containing
-            // all the column names and their types. This is used to create the model, and train it. 
+            // <Snippet11>
+            Evaluate(mlContext, model);
+            // </Snippet11>
+            
+            // <Snippet16>
+            Predict(mlContext, model);
+            // </Snippet16>
+
+            // <Snippet21>
+            TrainFinalModel(mlContext);
+            // </Snippet21>
+
+            // <Snippet25>
+            PredictWithModelLoadedFromFile(mlContext);
+            // <Snippet25>
+
+            Console.WriteLine();
+            Console.WriteLine("=============== End of process ===============");
+        }
+
+        public static ITransformer Train(MLContext mlContext, string dataPath)
+        {
+            //Note that this case, loading your training data from a file, 
+            //is the easiest way to get started, but ML.NET also allows you 
+            //to load data from databases or in-memory collections.
             // <Snippet6>
-            pipeline.Add(new TextLoader(_dataPath).CreateFrom<SentimentData>());
+            IDataView dataView =_textLoader.Read(new MultiFileSource(dataPath));
```

Comment:
>new MultiFileSource(dataPath) [](start = 49, length = 29)

let's use dataPath directly for the simple case of loading from a single path. MultiFileSource is not needed. Applies to most cases.